### PR TITLE
style(router/compat): small style clean

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

This PR is a small style clean for #10615.

### Checklist

- [ ] The Pull Request has tests
- [ ] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### CODE REVIEW

File: kong/router/compat.lua [hunk 1]
```diff
@@ -330,38 +332,44 @@ end
...
 
 local function split_routes_and_services_by_path(routes_and_services)
   local routes_and_services_split = tb_new(#routes_and_services, 0)
-  for i = 1, #routes_and_services do
-    split_route_by_path_into(routes_and_services[i], routes_and_services_split)
+
+  for _, v in ipairs(routes_and_services) do
+    split_route_by_path_into(v, routes_and_services_split)
   end
+
```
Comment (bungle) [inline kong/router/compat.lua:372]: why this change?

Comment (chronolaw) [reply kong/router/compat.lua:372]: I think `ipairs` may be better optimized by LuaJIT.

**Review by bungle — APPROVED** [review]
